### ISSUE TITLE
请求支持zxinc、ip2location lite离线库解析

### ISSUE THREAD

**Issue by famuserpro** [sjlleo/nexttrace#68]
是否可以添加对这些的支持，确保数据本地处理

**Comment by m-cameron** [comment]
IP2Location also has web service at ip2location.io for free usage.

**Comment by sjlleo** [comment]
> IP2Location also has web service at ip2location.io for free usage.

That's true. I can add it to one of the data sources in the NextTrace if you like.

However, I still need to determine if ip2location.io API allows the act of providing geolocation information to the user publicly, and if it needs to display IP2Location copyright information.

**Comment by famuserpro** [comment]
> IP2Location also has web service at ip2location.io for free usage.

Thanks, but due to the need for privacy protection, I don't want anyone to know my test results, even open source official data sources I don't want to use, so an offline version is necessary.

**Comment by tsosunchia** [comment]
感谢您的使用，我们已经在努力尝试添加对这两个数据库的支持。
但有一点需要特别说明NextTrace很多功能都基于我们独有的geoip api实现，当NextTrace配合第三方数据库时可能效果无法达到完美。

**Comment by famuserpro** [comment]
> 感谢您的使用，我们已经在努力尝试添加对这两个数据库的支持。 但有一点需要特别说明NextTrace很多功能都基于我们独有的geoip api实现，当NextTrace配合第三方数据库时可能效果无法达到完美。

nexttrace是一个不错的项目，但是刚刚提到的”我们独有的geoip api实现“是指哪些，我还是希望了解一下，谢谢。

**Comment by tsosunchia** [comment]
> > 感谢您的使用，我们已经在努力尝试添加对这两个数据库的支持。 但有一点需要特别说明NextTrace很多功能都基于我们独有的geoip api实现，当NextTrace配合第三方数据库时可能效果无法达到完美。
> 
> 
> 
> nexttrace是一个不错的项目，但是刚刚提到的”我们独有的geoip api实现“是指哪些，我还是希望了解一下，谢谢。

现版本的NextTrace是支持很多第三方库的，您可以通过-d参数引用。
当您多对比通过默认api和第三库输出的结果差异，即可回答您的问题。

**Comment by famuserpro** [comment]
感谢解答，如果不能显示这些信息的话，我希望可以隐藏这些部分空白

**Comment by tsosunchia** [comment]
> 感谢解答，如果不能显示这些信息的话，我希望可以隐藏这些部分空白

您提出的确实是一个好问题，但这牵扯到很大的重构工作量，我们将在合适的时机进行改进，谢谢。

**Comment by famuserpro** [comment]
我看官方的源中在ipv6基本都能提供地理位置，而且位置精确，有点出乎意料。我对比了很多库，发现并没有这部分信息，能否告知一下来源，感谢。

**Comment by tsosunchia** [comment]
> 我看官方的源中在ipv6基本都能提供地理位置，而且位置精确，有点出乎意料。我对比了很多库，发现并没有这部分信息，能否告知一下来源，感谢。

这也是我提到的 我们的特色功能之一 在这方便，我们下了很大努力 去人工校准，但是数据准确度离我们的目标还有很远。
您测了几个发现准，可能恰好是我们校准过的。

**Comment by famuserpro** [comment]
我测试的应该都算是冷门线路，每个地理位置都正确的显示了，一直到国内电信v6，没有问题，目前没有任何一个工具能做到这样的效果，看起来你们志愿者团队应该维护了一个相当大的库。

**Comment by tsosunchia** [comment]
> 是否可以添加对这些的支持，确保数据本地处理

https://github.com/sjlleo/nexttrace/releases/tag/v0.2.2
在早期版本NextTrace有支持`ip2region`，您可以先用这个来过渡。
